src/lfortran/semantics/ast_common_visitor.h

```diff
@@ -1250,6 +1250,71 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                                                 func_sym, func_sym, nullptr, 0, ASRUtils::symbol_type(end_sym), nullptr, nullptr, false));
                         end = func_call;
                     }
+                } else if(ASR::is_a<ASR::IntegerBinOp_t>(*end)) {
```

Comment:
```suggestion
                } else if(end && ASR::is_a<ASR::IntegerBinOp_t>(*end)) {
```

this shall do the trick.

Reply:
Pleas see https://github.com/lfortran/lfortran/pull/4034#issuecomment-2119174802